production/scrapers/south_carolina_population.R

```diff
@@ -0,0 +1,146 @@
+source("./R/utilities.R")
+
+south_carolina_population_restruct <- function(x, exp_date = Sys.Date()){
+    
+    # Check date 
+    pdf_date <- x %>% 
+        magick::image_read_pdf() %>% 
+        magick::image_crop("340x150+1900+100") %>% 
+        magick::image_ocr() %>% 
+        lubridate::mdy()
+    
+    error_on_date(pdf_date, exp_date)
+    
+    # Check header (first row)
+    exp_header <- c("Institution or Center", "General Housing", 
+                    "Restrictive Housing", "Programs", "Total")
+    
+    header <- x %>% 
+        magick::image_read_pdf() %>% 
+        magick::image_crop("2100x100+100+230") %>% 
+        magick::image_ocr() %>% 
+        stringr::str_split("\\|") %>% 
+        .[[1]] %>% 
+        matrix(nrow = 1, ncol = 5) %>% 
+        as.data.frame() %>% 
+        janitor::row_to_names(row_number = 1) 
+    
+    check_names(header, exp_header)
+    
+    # Check column names (row below header)
+    x_ <- x %>% 
+        magick::image_read_pdf() %>% 
+        magick::image_crop("3000x2200+0+300") %>% 
+        magick::image_convert(type = 'Bilevel') %>% 
+        ExtractTable() %>% 
+        as.data.frame()
+    
+    col_name_mat <- matrix(c(
+        "Name", "X0", "", 
+        "Gen.OpCap.Drop", "X1", "Operating Capacity",
+        "Gen.PhysCount.Drop", "X2", "Physical Count",
+        "Gen.OOS.Drop", "X3", "Out of Service",
+        "Gen.UR.Drop", "X4", "Utilization Rate",
+        "RH.OpCap.Drop", "X5", "Operating Capacity",
+        "RH.PhysCount.Drop", "X6", "Physical Count",
+        "RH.OOS.Drop", "X7", "Out of Service",
+        "RH.UR.Drop", "X8", "Utilization Rate",
+        "Prog.OpCap.Drop", "X9", "Operating Capacity",
+        "Prog.PhysCount.Drop", "X10", "Physical Count",
+        "Prog.OOS.Drop", "X11", "Out of Service",
+        "Prog.UR.Drop", "X12", "Utilization Rate",
+        "Total.OpCap.Drop", "X13", "Operating Capacity",
+        "Residents.Population", "X14", "Physical Count",
+        "Total.OOS.Drop", "X15", "Out of Service",
+        "Total.UR.Drop", "X16", "Utilization Rate",
+        "Filled.Triple.Drop", "X17", "Number of Filled Triple Cells"
+    ), ncol = 3, nrow = 18, byrow = TRUE)
+    
+    colnames(col_name_mat) <- c("clean", "raw", "check")
+    col_name_df <- as_tibble(col_name_mat)
+    
+    check_names_extractable(x_, col_name_df)
+    rename_extractable(x_, col_name_df) 
+}
+
+south_carolina_population_extract <- function(x){
+    x %>% 
+        select(!ends_with(".Drop")) %>% 
+        mutate(Name = clean_fac_col_txt(Name, to_upper = TRUE)) %>% 
+        filter(!str_detect(Name, "TOTAL")) %>% 
+        filter(!Residents.Population %in% c("Physical Count", ""))  %>% 
+        mutate(Name = ifelse(str_detect(Name, "KIRKLAND"), "KIRKLAND", Name), 
+               Name = ifelse(str_detect(Name, "GRAHAM"), "GRAHAM", Name)) %>% 
+        clean_scraped_df() %>% 
+        group_by(Name) %>% 
+        summarise(Residents.Population = sum(Residents.Population)) %>% 
```

Comment:
It looks like Residents.Population shouldn't ever be NA, but just to be safe, you might consider adding in `sum(Residents.Population, na.rm = TRUE)`

Reply:
That's a super, super good call – fixed!